src/libserver/session.cpp

```diff
@@ -1684,24 +1688,48 @@ JsonApiResult Session::callListingsJsonApi(
 	JsonApiMethod method, const QStringList &path, const QJsonObject &request)
 {
 	Q_UNUSED(request);
-	if(path.length() != 1)
+
+	int pathLength = path.length();
+
+	if(pathLength > 1)
 		return JsonApiNotFound();
-	const int id = path.at(0).toInt();
 
-	const auto announcements = m_announcements->getAnnouncements(this);
-	for(const sessionlisting::Announcement &a : announcements) {
-		if(a.listingId == id) {
-			if(method == JsonApiMethod::Delete) {
-				unlistAnnouncement(a.apiUrl.toString());
-				return JsonApiResult{
-					JsonApiResult::Ok,
-					QJsonDocument(QJsonObject{{"status", "ok "}})};
+	if(pathLength == 1) {
+		const int id = path.at(0).toInt();
 
-			} else {
-				return JsonApiBadMethod();
+		const auto announcements = m_announcements->getAnnouncements(this);
+		for(const sessionlisting::Announcement &a : announcements) {
+			if(a.listingId == id) {
+				if(method == JsonApiMethod::Delete) {
+					unlistAnnouncement(a.apiUrl.toString());
+					return JsonApiResult{
+						JsonApiResult::Ok,
+						QJsonDocument(QJsonObject{{"status", "ok "}})};
+
+				} else {
+					return JsonApiBadMethod();
+				}
 			}
 		}
+
+		return JsonApiNotFound();
+	}
+
+	if (method == JsonApiMethod::Get) {
+		return JsonApiResult{
+			JsonApiResult::Ok, QJsonDocument(getListingsDescription())};
+			
+	} else if(method == JsonApiMethod::Create) {
+		if(request.contains("url")) {
+			makeAnnouncement(QUrl(request["url"].toString()));
+			return JsonApiResult{
+				JsonApiResult::Ok,
+				QJsonDocument(QJsonObject{{"status", "ok "}})};
+		}
```

Comment:
If the URL is missing, this should return a 400 error, not fall through to a 404. Something like:

```
} else {
    return JsonApiErrorResult(JsonApiResult::BadRequest, QStringLiteral("Missing url"));
}
```